src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResource.java

```diff
@@ -414,45 +414,48 @@ public Response postProgressReport(
     try {
       User user = userService.findUserByEmail(authUser.getEmail());
       DataAccessRequest parentDar = dataAccessRequestService.findByReferenceId(parentReferenceId);
-      if (!user.getUserId().equals(parentDar.getUserId())) {
-        throw new ForbiddenException("User not authorized to update this Data Access Request");
-      }
-
-      DataAccessRequest payload = populateDarFromJsonString(user, dar);
-      DataAccessRequest childDar = dataAccessRequestService.createDataAccessRequest(user, payload);
+      DataAccessRequest payload = populateProgressReportFromJsonString(dar, parentDar);
+      updateProgressReportWithDocuments(collabInputStream, collabFileDetails, ethicsInputStream,
+          ethicsFileDetails, payload, parentDar);
+      DataAccessRequest progressReport = dataAccessRequestService.createProgressReport(user, payload, parentDar);
+      return Response.ok(progressReport.convertToSimplifiedDar()).build();
+    } catch (Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
 
-      for (Integer datasetId : childDar.getDatasetIds()) {
-        Dataset dataset = datasetService.findDatasetById(datasetId);
-        if (dataset == null) {
-          throw new NotFoundException("Dataset " + datasetId + " not found");
-        }
-        DataUse dataUse = dataset.getDataUse();
-        if (dataUse == null || dataUse.getCollaboratorRequired() == null
-            || dataUse.getEthicsApprovalRequired() == null) {
-          throw new BadRequestException("Dataset " + datasetId + " is missing data use(s)");
-        }
-        if (dataUse.getCollaboratorRequired()) {
-          String parentCollabLocation = parentDar.getData().getCollaborationLetterLocation();
-          if ((collabFileDetails == null || collabFileDetails.getSize() <= 0)
-              && Strings.isNullOrEmpty(parentCollabLocation)) {
-            throw new BadRequestException("Collaboration document is required");
-          }
-          childDar = updateDarWithDocumentContents(DarDocumentType.COLLABORATION, user, childDar,
-              collabInputStream, collabFileDetails);
+  public void updateProgressReportWithDocuments(InputStream collabInputStream,
+      FormDataContentDisposition collabFileDetails, InputStream ethicsInputStream,
+      FormDataContentDisposition ethicsFileDetails, DataAccessRequest childDar,
+      DataAccessRequest parentDar) throws IOException {
+    for (Integer datasetId : childDar.getDatasetIds()) {
+      Dataset dataset = datasetService.findDatasetById(datasetId);
+      if (dataset == null) {
+        throw new NotFoundException("Dataset " + datasetId + " not found");
+      }
```

Comment:
Probably scope creep here, but `datasetService.findDatasetById` really should throw the NFE. We do this a lot in the service layer now, not sure why this method doesn't do that. I think a follow-on ticket would be nice for this.